### DISCUSSION TITLE
Image Pull Secrets for Jobs

### DISCUSSION THREAD

**Discussion by akath19** [aquasecurity/starboard#169] [Questions & Help] [answered]
Hi,

I've been testing Starboard on GKE with [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) (basically a way of automatically authenticating workloads against GCP via annotations on ServiceAccounts) for a few days and after a lot of debugging I have a few questions that might end up being a bug or enhancement:

_Issue:_ When starboard is installed (via `init`) the starboard ServiceAccount is created in the starboard namespace, however, it isn't used at all for running jobs, the default ServiceAccount is used instead, I found this after digging through the code to find [this](https://github.com/aquasecurity/starboard/blob/4aa0d5259b76711a8053bb4fe2ce31bf89299c20/pkg/kube/pod/manager.go#L21), this leads me to a couple of questions

_Questions_: 

1. Should the starboard ServiceAccount be used as a centralized config for ImagePullSecrets?
2. Why do all jobs run with the default ServiceAccount in the `starboard `namespace
3. Should the ServiceAccount a job runs as be configurable?

**Comment by danielpacak** [comment, marked as answer]
👋 Thank you for your questions @akath19

> 1. Should the starboard ServiceAccount be used as a centralized config for ImagePullSecrets?

IIRC in v0.4.0 we do not use `starboard` ServiceAccount consistently. However, it's been fixed on the main branch so each scanner runs as `starboard` ServiceAccount. I did check that, but if it's not the case please file an issue.

> 2. Why do all jobs run with the default ServiceAccount in the starboard namespace

It's our legacy when we prototyped Starboard

> 3. Should the ServiceAccount a job runs as be configurable

Since the init command creates the `starboard` ServiceAccount and then it's used in a few places in the code already, I'd not make it configurable. However, if we collect some real use cases and prove it's valuable we can always do that.

**Comment by akath19** [comment]
Awesome, I noticed that it's already fixed on main but I wasn't sure if it was going to stay that way, thanks!
